### PR TITLE
JIT: Add out-of-bounds fallback for AdvSimd.ShiftRightLogical

### LINKED ISSUE #105621: JIT: `AdvSimd.ShiftRightLogical(x, 0)` throws in debug but not in release
```csharp
// Generated by Fuzzlyn v2.1 on 2024-07-28 20:49:49
// Run on Arm64 Linux
// Seed: 14204794442367797079-vectort,vector64,vector128,armadvsimd,armadvsimdarm64,armaes,armarmbase,armarmbasearm64,armcrc32,armcrc32arm64,armdp,armrdm,armrdmarm64,armsha1,armsha256
// Reduced from 12.6 KiB to 0.4 KiB in 00:00:17
// Debug: Throws 'System.ArgumentOutOfRangeException'
// Release: Runs successfully
using System;
using System.Runtime.CompilerServices;
using System.Numerics;
using System.Runtime.Intrinsics;
using System.Runtime.Intrinsics.Arm;

public class Program
{
    public static void Main()
    {
        var vr3 = Vector64.Create<byte>(0);
        var vr4 = AdvSimd.ShiftRightLogical(vr3, 0);
        System.Console.WriteLine(vr4);
    }
}
```

cc @dotnet/jit-contrib 

This is with #105527 included. Seems like it gets constant folded even though it should throw.

### PR DESCRIPTION
**Pull request by amanasifkhalid**
Fixes #105621. When its immediate is out-of-bounds, `AdvSimd.ShiftRightLogical` can be transformed into `AdvSimd.ShiftLogical`, which takes the immediate in a register. This means `AdvSimd.ShiftRightLogical` will no longer throw `ArgumentOutOfRangeException` in Debug or Release; when optimizing, we'd previously fold the intrinsic away, thus creating behavioral discrepancies.

### CODE REVIEW

**Comment by amanasifkhalid** [comment]
cc @dotnet/jit-contrib, @tannergooding PTAL.

**Review by tannergooding — APPROVED** [review]

File: src/coreclr/jit/hwintrinsicarm64.cpp [hunk 1]
```diff
@@ -564,7 +564,29 @@ void HWIntrinsicInfo::lookupImmBounds(
...
+
+            GenTree* op2 = impPopStack().val;
+            GenTree* op1 = impSIMDPopStack();
+
+            // AdvSimd.ShiftLogical does right-shifts with negative immediates, hence the negation
+            GenTree* tmpOp =
+                gtNewSimdCreateBroadcastNode(simdType, gtNewOperNode(GT_NEG, genActualType(op2->TypeGet()), op2),
+                                             simdBaseJitType, genTypeSize(simdType));
+            return gtNewSimdHWIntrinsicNode(simdType, op1, tmpOp, NI_AdvSimd_ShiftLogical, simdBaseJitType,
+                                            genTypeSize(simdType));
```
Comment (jakobbotsch) [inline src/coreclr/jit/hwintrinsicarm64.cpp:584]: Is the spilling necessary?

File: src/tests/JIT/Regression/JitBlue/Runtime_105621/Runtime_105621.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
```
Comment (EgorBo) [inline src/tests/JIT/Regression/JitBlue/Runtime_105621/Runtime_105621.cs:3]: We usually copy-paste Fuzzlyn's header for repros

File: src/coreclr/jit/hwintrinsicarm64.cpp (hunk 1 above)
Comment (EgorBo) [reply src/coreclr/jit/hwintrinsicarm64.cpp]: yep, we don't. Here op1 is still evaluated before op2 so all good

**Review by EgorBo — APPROVED** [review]

**Pull request merged**